### PR TITLE
feat(fe-fpm-writer): Custom header section

### PR DESCRIPTION
**Pull request by broksy**
Added custom header section for `fe-fpm-writer`.

### CODE REVIEW

File: packages/fe-fpm-writer/src/section/types.ts [hunk 1]
```diff
@@ -27,6 +27,13 @@ export interface InternalCustomSection extends CustomSection, CustomSubSection,
     dependencies?: string;
 }
 
+export interface CustomHeaderSection extends CustomSection {
+    /**
+     * Sub title of header section.
+     */
+    subTitle?: string;
```
Comment (815are) [inline packages/fe-fpm-writer/src/section/types.ts:34]: what about other properties?
Like `templateEdit`, `stashed` , `requestGroupId`, `flexSettings`

More details about properties can be find here - https://ui5.sap.com/test-resources/sap/fe/core/fpmExplorer/index.html#/customElements/customHeaderFacetContent
I see following are missing:
image

**Review by 815are — CHANGES_REQUESTED** [review]

File: packages/fe-fpm-writer/src/section/index.ts [hunk 2]
```diff
@@ -113,7 +113,42 @@ function generate(
...
+export function generateCustomHeaderSection(
+    basePath: string,
+    customHeaderSection: CustomHeaderSection,
+    fs?: Editor
+): Editor {
+    const manifestRoot = getManifestRoot('headersection', customHeaderSection.minUI5Version);
+    const { editor, section } = generate(basePath, customHeaderSection, manifestRoot, fs);
+    // handle fragment for templateEdit property
+    if (section.edit) {
+        const viewPath = join(section.path, `${section.edit.name}.fragment.xml`);
```
Comment (815are) [inline packages/fe-fpm-writer/src/section/index.ts:136]: Here code calculates `viewPath` for `templateEdit` property based on `config.edit`/`sections.edit`.
I think `section.path` is wrong and `path` should be calculated based on `config.edit`/`sections.edit`.
Here is generic code used to calculate `path` -> https://github.com/SAP/open-ux-tools/blob/main/packages/fe-fpm-writer/src/common/defaults.ts#L23

If we follow same logic for `templateEdit`, then same logic should be applied for `section.edit`, I mean something like:
```
setCommonDefaults(section.edit, manifestPath, manifest)
```

File: packages/fe-fpm-writer/test/unit/headersection.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    // Basic custom header section object
+    const customHeaderSection: CustomHeaderSection = {
+        target: 'sample',
+        name: 'NewCustomHeaderSection',
+        folder: 'extensions/custom',
+        title: 'New Custom Header Section',
+        subTitle: 'Custom Header section subtitle',
+        edit: {
+            name: 'NewCustomHeaderSectionEdit',
+            folder: 'extensions/custom',
```
Comment (815are) [inline packages/fe-fpm-writer/test/unit/headersection.test.ts:26]: I think there is missing test where folders are different  `folder` vs `edit.folder`, which associated to comment https://github.com/SAP/open-ux-tools/pull/1142/files#r1285789765

File: packages/fe-fpm-writer/src/index.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export { CustomTableColumn } from './column/types';
 export { generateCustomColumn } from './column';
 
 export { CustomSection, CustomSubSection } from './section/types';
```
Comment (815are) [inline packages/fe-fpm-writer/src/index.ts:10]: I think additional exports are missing, like `CustomHeaderSection` and enums `RequestGroupId`, `DesignTime`

**Review by 815are — COMMENTED** [review]
@broksy 

Missing following parts:
1. changeset is missing
2. If we follow other extension's approach then additional test is missing in integration test -> https://github.com/SAP/open-ux-tools/blob/feat/customHeaderSection/packages/fe-fpm-writer/test/integration/basic-lrop-app.test.ts#L299

File: packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap [hunk 5]
```diff
@@ -54,6 +54,52 @@ Object {
...
+    "state": "modified",
+  },
+  "js/webapp/ext/myCustomHeaderSectionEdit/MyCustomHeaderSectionEdit.js": Object {
+    "contents": "sap.ui.define([
+    \\"sap/m/MessageToast\\"
+], function(MessageToast) {
+    'use strict';
+
+    return {
+        onPress: function(oEvent) {
```
Comment (815are) [inline packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap:98]: could we please rename `onPress` to `onChange` if input is used and event is `change`?

Comment (broksy) [reply packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap]: You're right! Changed to `onChange`.

File: packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap [hunk 6]
```diff
@@ -450,6 +496,21 @@ Object {
                       \\"enabled\\": true,
                       \\"text\\": \\"Navigate to CustomPage (appended action)\\"
                     }
+                  },
+                  \\"facets\\": {
+                    \\"MyCustomHeaderSection\\": {
+                      \\"template\\": \\"v4travel.ext.myCustomHeaderSection.MyCustomHeaderSection\\",
+                      \\"templateEdit\\": \\".MyCustomHeaderSectionEdit\\",
```
Comment (815are) [inline packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap:503]: `templateEdit` value looks wrong. I think it should be `v4travel.ext.myCustomHeaderSectionEdit.MyCustomHeaderSectionEdit`

File: packages/fe-fpm-writer/test/unit/__snapshots__/header-section.test.ts.snap [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+  },
+  "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
+  "position": Object {
+    "anchor": "Dummy",
+    "placement": "Before",
+  },
+  "requestGroupId": "Decoration",
+  "subTitle": "Custom Header section subtitle",
+  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
+  "templateEdit": ".NewCustomHeaderSectionEdit",
```
Comment (815are) [inline packages/fe-fpm-writer/test/unit/__snapshots__/header-section.test.ts.snap:16]: same issue like mentioned above in intergration test snapshot. Looks like wrongly generated entry for `templateEdit` - it does not have namespace before fragment name

**Review by 815are — CHANGES_REQUESTED** [review]

File: packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap (hunk 6 above)
Comment (815are) [reply packages/fe-fpm-writer/test/integration/__snapshots__/basic-lrop-app.test.ts.snap]: fix - 4c6fcdda05fc483345b76f297408a89e82084633

File: packages/fe-fpm-writer/test/unit/__snapshots__/header-section.test.ts.snap (hunk 7 above)
Comment (815are) [reply packages/fe-fpm-writer/test/unit/__snapshots__/header-section.test.ts.snap]: fix - 4c6fcdda05fc483345b76f297408a89e82084633

**Review by 815are — APPROVED** [review]
Checked:
1. Code looks correct and easy to understand;
2. Tests are maintained and new multiple scenarios regarding `templateEdit` are added;
3. Changeset is presented;

I did not test new api method manually with real app.

Minor question regarding properties `requestGroupId` and `flexSettings` - if they are required or can be optional ->

**Review by zdravko-georgiev — APPROVED** [review]
Code looks good and it is easy to understand
Very good test coverage
Didn't tested manually

**Pull request merged**